### PR TITLE
refactor(connector): [PowerTranz] refactor powertranz payments to remove default cases

### PR DESCRIPTION
**Pull request by Himanshu-370**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/powertranz/transformers.rs [hunk 1]
```diff
@@ -101,9 +102,24 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for PowertranzPaymentsRequest
     fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
         let source = match item.request.payment_method_data.clone() {
             api::PaymentMethodData::Card(card) => Ok(Source::from(&card)),
-            _ => Err(errors::ConnectorError::NotImplemented(
-                "Payment method".to_string(),
-            )),
+            api::PaymentMethodData::CardRedirect(_)
```
Comment (prasunna09) [inline crates/router/src/connector/powertranz/transformers.rs:105]:
```suggestion
            api::PaymentMethodData::Wallet(_)
            | api::PaymentMethodData::CardRedirect(_)
```
Here you haven't include wallet enum. This is the reason for clippy check to fail.

File: crates/router/src/connector/powertranz/transformers.rs [hunk 2]
```diff
@@ -101,9 +102,24 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for PowertranzPaymentsRequest
...
+            | api::PaymentMethodData::Upi(_)
+            | api::PaymentMethodData::Voucher(_)
+            | api::PaymentMethodData::GiftCard(_) => Err(errors::ConnectorError::NotSupported {
+                message: utils::SELECTED_PAYMENT_METHOD.to_string(),
+                connector: "powertranz",
+            })
+            .into_report(),
+            // _ => Err(errors::ConnectorError::NotImplemented(
+            //     "Payment method".to_string(),
+            // )),
```
Comment (prasunna09) [inline crates/router/src/connector/powertranz/transformers.rs:122]:
```suggestion
```
Please remove these comments

**Review by prasunna09 — CHANGES_REQUESTED** [review]

**Review by srujanchikke — APPROVED** [review]
LGTM!

**Review by prasunna09 — APPROVED** [review]
LGTM! Thank you for you contribution. @Himanshu-370

**Review by SanchithHegde — APPROVED** [review]

**Comment by swangi-kumari** [comment]
Hey @Himanshu-370 ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
Thanks!

**Pull request merged**